sdk/nodejs/cloudwatch/metric.ts

```diff
@@ -0,0 +1,402 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as pulumi from "@pulumi/pulumi";
+
+import * as alarm from "./metricAlarm";
+import * as sns from "../sns";
+import * as utils from "../utils";
+
+/**
+ * Metrics are the fundamental concept in CloudWatch. A metric represents a time-ordered set of data
+ * points that are published to CloudWatch. Think of a metric as a variable to monitor, and the data
+ * points as representing the values of that variable over time. For example, the CPU usage of a
+ * particular EC2 instance is one metric provided by Amazon EC2. The data points themselves can come
+ * from any application or business activity from which you collect data.
+ *
+ * AWS services send metrics to CloudWatch, and you can send your own custom metrics to CloudWatch.
+ * You can add the data points in any order, and at any rate you choose. You can retrieve statistics
+ * about those data points as an ordered set of time-series data.
+ *
+ * Metrics exist only in the region in which they are created. Metrics cannot be deleted, but they
```

Comment:
If a metric cannot be deleted, how will Pulumi handle this?

Reply:
discussed offline.